### DISCUSSION TITLE
Disable mouse on GroupBox

### DISCUSSION THREAD

**Discussion by gruvw** [qtile/qtile#4183] [Q&A]
Hello, I found it annoying to have mouse actions on the `GroupBox` widget (placed inside `bar.Bar`).  
Is there a way to disable mouse related events on the `GroupBox` (or even better, on the whole `bar.Bar`, except `Systray` of course) ?  
By mouse related events, I include clicks, but also scrolling (which cycles through groups).

Thanks a lot!

**Comment by elParaguayo** [comment]
Which aspects of the mouse actions are you finding annoying? For `GroupBox`, if it's the dragging of groups, you can disable this by setting `disable_drag=True` in the widget config.

If you want to disable mouse actions for widgets, one way to do this would be to have a function (hooked to startup or config reload) that loops over widgets in the bar and sets the `self.mouse_callbacks` attribute to an empty dict.

**Reply by gruvw** [reply]
I already have `disable_drag=True` but it's the mouse click that switches groups and the scrolling that switches groups that annoys me.

**Reply by m-col** [reply]
I think it would be OK to add `interactive=True` as a default option on one of the base widget classes, then use it to conditionally set `mouse_callbacks` within `add_callbacks`. PR welcome :)

or another similarly hacky solution might be (untested):
```py
from libqtile.widget.base import _Widget

_Widget.add_callbacks = lambda *args: None
```

**Reply by gruvw** [reply]
Should we convert this into an issue then ?
